docs/adr/012_decouple_from_edge.md

```diff
@@ -52,77 +52,105 @@ The main difference comes to "ModuleConnectionHost", which is responsible for:
+A disadvantage, though, comes from the request quota limits on IoT Hub Device Registry operations which could be cause of throttling and instability for other pieces of the LNS itself.
+
+#### Direct method invocation
 
 As previously mentioned, IoT Edge is allowing the functionality of "direct method invocations" giving the possiblity to remotely invoke some functions on the LNS.
 
-When decoupling LNS from Azure IoT Edge, there are two possibilities available for handling the scenarios covered by direct method invocation.
+When decoupling LNS from Azure IoT Edge, there are different possibilities for handling the scenarios covered by direct method invocation.
+
+As stated in previous paragraphs, the "Facade" Azure Function has to differentiate between LNS deployment modes depending on the "LNS device id".
 
-For both mechanisms the "Facade" Azure Function bundled with the Starter Kit, has to differentiate between a "edge" or "cloud" deployment of LNS. The proposal is to **differentiate the mechanism used for remotely invoke methods depending on the "LNS device id"**. This means that an Azure IoT Edge device will still be using the direct method invocations provided by IoT Hub whereas the "cloud" deployments will use a new, different, mechanism.
+This means that an Azure IoT Edge device will still be using the direct method invocations provided by IoT Hub whereas the "cloud" deployments will use a new, different, mechanism.
 
-#### Expose REST endpoints on the LNS
+For "cloud" deployments, such mechanism consists in using Redis Pub/Sub to publish the method invocation events on a topic, to which all the LNS instances are subscribing and filtering based on their "hostname".
 
-One alternative might be to expose some REST endpoints on the LNS. This is the easiest approach to take and would simplify the "manual" invocation of such methods, even though this is not in scope of this document.
+The decision is to:
 
-The scenarios covered by these method invocations are meant to target exactly one desired LNS instance and not any of them.
+- Reuse [Azure Cache for Redis](https://docs.microsoft.com/en-us/azure/azure-cache-for-redis/cache-overview) instance that it's already being deployed in the solution as it allows the usage of [Redis Pub/Sub functionality](https://redis.io/docs/manual/pubsub/)
+- As a stretch goal, useful for high scale and testing scenarios, implement a new function for clearing cache of all the LNS subscribed to the topic
 
-As an example, let's assume the scenario where we want to send a cloud-to-device message to a specific class-c leaf device. In this scenario we need to target the exact LNS which is managing the connection with the LoRa Basics Station in reach of the class-c device we want to target.
+An alternative to "Azure Cache for Redis" for the Pub/Sub functionality is to use "Azure Service Bus". While this would allow higher resiliency, the alternative has been discarded to avoid provisioning another additional service.
```

Comment:
as long as we abstract the pub/sub implementation, the actual service should be transparent.

Reply:
Agree, will clarify that the goal is abstracting it but still choose Redis in this phase

Reply:
Done in b539b62d586c7ec203a11ee4490ed7e2511da5d7